### ISSUE TITLE
OS package libffi needed for building some Python packages

### ISSUE THREAD

**Issue by gtoffoli** [up2university/kuboxed#25]
I wasn't able to install the _Scrapy_ python library, for running the notebook _Using Scrapy in Jupyter notebook_; see: https://www.jitsejan.com/using-scrapy-in-jupyter-notebook.html.
The error log, attached below, includes a lot of error messages `No package 'libffi' found`; see: https://sourceware.org/libffi/
It seems that the OS package _libffi_, and possibly also _libffi-dev_, is needed to build some python packages on Linux.

[import Scrapy.txt](https://github.com/up2university/kuboxed/files/2657925/import.Scrapy.txt)

**Comment by diocas** [comment]
Hi Giovanni,
Sorry for the late reply but we were upgrading SWAN. For now we have the new version in swan.test.up2university.eu.
I've installed the packages for the Shakespeare (that you had already done manually) and also installed Scrapy.
Can you please check if you can run your notebook? Is there any other you would like to run and that needs some other packages?
Cheers,
Diogo

**Comment by gtoffoli** [comment]
Hi diogo,
thank you, but I can't access swan.test.up2university.eu
I can access the production instance swan.up2university.eu using my Up2U-SSO account.
If I try to access the test instance in the same way, I get the "403 : Forbidden - You don't seem to have permission to use SWAN. Did you create a CERNBox account? You need to do it first". If I follow the link, I get a login form from cernbox.test.up2university.eu

**Comment by diocas** [comment]
Hi Giovanni,
That is the expected behaviour. To use SWAN you need a CERNBox folder, so you need to visit the CBox page at least once (that will create your folder). After you login to CBox, you can then open SWAN and it should open.
But you're saying that when you visit cernbox.test.up2university.eu it doesn't log you in? Can you try going to that address in a private window?
Thanks

**Comment by gtoffoli** [comment]
Hi Diogo,
I've visited cernbox.test.up2university.eu in a private window, with both FF and Chrome; after I authenticate with the Up2U Login, I am redirected to https://cernbox.test.up2university.eu/cernbox/index.php/login
But I haven't credential to locally login; and if at this point I try to visit swan.test.up2university.eu I get again the "403 : Forbidden" response.

**Comment by diocas** [comment]
Hi Giovanni,
Your user had a problem. I removed the old data and now it should work.
Please give it a try and let me know.
Sorry for the troubles.

**Comment by gtoffoli** [comment]
Hi Diogo,
thank you: now I can access SWAN and CERNBox on the test instance.
I was able to import scrapy, but only after installing it with the command _pip install --user scrapy_
Then I was able to copy and execute a notebook for testing scrapy : _Using Scrapy in Jupyter notebook_ (https://www.jitsejan.com/using-scrapy-in-jupyter-notebook.html)
I was confused a little by the new menu and folders layout in the UI.
A couple of problems:
- cannot logout: I always get a System error
- I wanted to configure a bash script to be executed at the start, but always get an error when trying to add a text file; the error trace ends with; 
```
  File "/usr/local/lib/python3.6/site-packages/notebook/services/contents/fileio.py", line 203, in open
    with io.open(os_path, *args, **kwargs) as f:
OSError: [Errno 101] Network is unreachable: '/eos/user/u/up001079/SWAN_projects/untitled.txt'

```

**Comment by diocas** [comment]
Hi Giovanni,
Thank you for testing.
The local installation of Scrapy is hidden from the user environment (but it shouldn't). I'll try to fix this for the next release, but I'm glad you managed to run it either way.

The logout problem is an ongoing one and we still don't have a solution for it: #12
The error creating a file is some problem in the storage. We will check and get back to you.

**Comment by diocas** [comment]
Hi Giovanni,
The error creating a file should be solved. I just updated a component in storage deployment that was very old. Can you try?
Thanks

**Comment by gtoffoli** [comment]
I tried. It's ok.
Thanks!
